bluemira/radiation_transport/radiation_profile.py

```diff
@@ -0,0 +1,2173 @@
+        }
+
+        self.t_and_n_pol = {}
+        for side, var in t_and_n_pol_inputs.items():
+            self.t_and_n_pol[side] = self.build_sector_distributions(**var)
+
+        return self.t_and_n_pol
+
+    def build_sol_radiation_distribution(
+        self,
+        t_and_n_pol_lfs,
+        t_and_n_pol_hfs,
+    ):
+        """
+        Radiation profiles builder.
+        For each scrape-off layer sector, it gives the
+        radiation profile along each flux tube.
+
+        Parameters
+        ----------
+        t_and_n_pol_lfs_low: array
+            temperature and density poloidal profile along each
+            flux tube within the lfs lower divertor set
+        t_and_n_pol_lfs_up: array
+            temperature and density poloidal profile along each
+            flux tube within the lfs upper divertor set
+        t_and_n_pol_hfs_low: array
+            temperature and density poloidal profile along each
+            flux tube within the hfs lower divertor set
+        t_and_n_pol_hfs_up: array
+            temperature and density poloidal profile along each
+            flux tube within the hfs upper divertor set
+
+        Returns
+        -------
+        rad["lfs_low"]: array
+            radiation poloidal profile along each
+            flux tube within the lfs lower divertor set
+        rad["lfs_up"]: array
+            radiation poloidal profile along each
+            flux tube within the lfs upper divertor set
+        rad["hfs_low"]: array
+            radiation poloidal profile along each
+            flux tube within the hfs lower divertor set
+        rad["hfs_up"]: array
+            radiation poloidal profile along each
+            flux tube within the hfs upper divertor set
+        """
+        # For each impurity species and for each flux tube,
+        # poloidal distribution of the radiative power loss function.
+        # Values along the open flux tubes
+        loss = {
+            "lfs": t_and_n_pol_lfs[0],
+            "hfs": t_and_n_pol_hfs[0],
+        }
+
+        for side, t_pol in loss.items():
+            loss[side] = [
+                [radiative_loss_function_values(t, t_ref, l_ref) for t in t_pol]
+                for t_ref, l_ref in zip(self.imp_data_t_ref, self.imp_data_l_ref)
+            ]
+
+        # For each impurity species and for each flux tube,
+        # poloidal distribution of the line radiation loss.
+        # Values along the open flux tubes
+        self.rad = {
+            "lfs": {"density": t_and_n_pol_lfs[1], "loss": loss["lfs"]},
+            "hfs": {"density": t_and_n_pol_hfs[1], "loss": loss["hfs"]},
+        }
+        for side, ft in self.rad.items():
+            self.rad[side] = [
+                [
+                    calculate_line_radiation_loss(n, l_f, fi)
+                    for n, l_f in zip(ft["density"], f)
+                ]
+                for f, fi in zip(ft["loss"], self.impurities_content)
+            ]
```

Comment:
bit weird to make a dictionary and then destroy
```suggestion
        rad_data = {
            "lfs": {"density": t_and_n_pol_lfs[1], "loss": loss["lfs"]},
            "hfs": {"density": t_and_n_pol_hfs[1], "loss": loss["hfs"]},
        }
        self.rad = {}
        for side, ft in rad_data.items():
            self.rad[side] = [
                [
                    calculate_line_radiation_loss(n, l_f, fi)
                    for n, l_f in zip(ft["density"], f)
                ]
                for f, fi in zip(ft["loss"], self.impurities_content)
            ]
```